client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -30,29 +31,32 @@
  */
 public final class Command {
 
-    private final boolean valid;
+    /**
+     * If present, the command is invalid.
+     */
+    private final Optional<String> validationError;
```

Comment:
is there a particular reason why you want to use an Optional here instead of just a simple String?

Reply:
That's just for making it more obvious that `validationError` may be not set, marking the important distinction between a valid and not valid Command.